src/elements/content-sidebar/activity-feed/annotations/AnnotationActivityLink.tsx

```diff
@@ -4,28 +4,45 @@ import noop from 'lodash/noop';
 import { injectIntl, MessageDescriptor, WrappedComponentProps } from 'react-intl';
 import PlainButton from '../../../../components/plain-button';
 import { ButtonType } from '../../../../components/button';
+import messages from './messages';
 import './AnnotationActivityLink.scss';
 
 type MessageDescriptorWithValues = {
-    values?: Record<string, number>;
+    values?: { number: string };
 } & MessageDescriptor;
 export interface AnnotationActivityLinkProps extends WrappedComponentProps {
     className?: string;
+    fileVersion: string;
     id: string;
-    isDisabled: boolean;
-    message: MessageDescriptorWithValues;
+    isCurrentVersion?: boolean;
+    locationValue: string;
     onClick: (id: string) => void;
+    shouldHideLink: boolean;
 }
 
 const AnnotationActivityLink = ({
     className,
     id,
+    fileVersion,
     intl,
-    isDisabled = false,
-    message,
+    isCurrentVersion,
+    locationValue,
     onClick = noop,
+    shouldHideLink,
     ...rest
-}: AnnotationActivityLinkProps): JSX.Element => {
+}: AnnotationActivityLinkProps): JSX.Element | null => {
+    if (shouldHideLink) {
+        return null;
+    }
+
+    const isFileVersionUnavailable = fileVersion == null;
+
+    const linkMessage = isCurrentVersion ? messages.annotationActivityPageItem : messages.annotationActivityVersionLink;
```

Comment:
Lines 40 - 45 build on each other, we could encapsulate them into a function. Might be easier to read with if statements and guard clauses too